python/lsst/source/injection/inject_base.py

```diff
@@ -0,0 +1,636 @@
+        grammar = "source" if len(sources_to_keep) == 1 else "sources"
+        self.log.info(
+            "Catalog cleaning removed %d of %d %s; %d remaining for catalog flag checking.",
+            num_cleaned_total,
+            len(sources_to_keep),
+            grammar,
+            np.sum(sources_to_keep),
+        )
+        injection_catalog = injection_catalog[sources_to_keep]
+        return injection_catalog
+
+    def _flag_sources(self, injection_catalog, input_exposure, binary_flags):
+        """Flag sources in the injection catalog prior to source injection.
+
+        This method will flag sources in the injection catalog prior to
+        injection. Checks will be made on a number of parameters, including
+        magnitude, source type and Sérsic index (where relevant).
+
+        Legacy profile types will be renamed to their standardized GalSim
+        equivalents; any source profile types that are not GalSim classes will
+        be flagged.
+
+        Note: Unlike the cleaning method, no sources are actually removed here.
+        Instead, a binary flag is set in the *injection_flag* column for each
+        source. Only unflagged sources will be generated for source injection.
+
+        Parameters
+        ----------
+        injection_catalog : `astropy.table.Table`
+            Catalog of sources to be injected.
+        input_exposure : `lsst.afw.image.exposure.ExposureF`
+            The exposure sources will be injected into.
+        binary_flags : `dict` [`str`, `int`]
+            Dictionary of binary flags to be used in the injection_flag column.
+
+        Returns
+        -------
+        injection_catalog : `astropy.table.Table`
+            The cleaned catalog of sources to be injected.
+        """
+        self.config = cast(BaseInjectConfig, self.config)
+
+        # Flag erroneous magnitude values (missing mag data or NaN mag values).
+        if "mag" not in injection_catalog.columns:
+            # Check injection_catalog has a mag column.
+            self.log.warning("No magnitude data found in injection catalog; cannot inject any sources!")
+            injection_catalog["injection_flag"] += 2 ** binary_flags["MAG_BAD"]
+        else:
+            # Check that all input mag values are finite.
+            mag_array = np.isfinite(ma.array(injection_catalog["mag"]))
+            bad_mag = ~(mag_array.data * ~mag_array.mask)
+            if (num_bad_mag := np.sum(bad_mag)) > 0:
+                grammar = "source" if num_bad_mag == 1 else "sources"
+                self.log.warning(
+                    "Flagging %d injection %s that do not have a finite magnitude.", num_bad_mag, grammar
+                )
+                injection_catalog["injection_flag"][bad_mag] += 2 ** binary_flags["MAG_BAD"]
+
+        # Replace legacy source types with standardized profile names.
+        injection_catalog["source_type"] = injection_catalog["source_type"].astype("O")
+        replace_dict = {
```

Comment:
Do we want to continue to support all of these in this refactor?  The only reasonable one (in my opinion) is star -> DeltaFunction.  I'd get rid of the rest.

Reply:
I'm split on this. On the one hand, keeping it in costs us little and may help legacy users in the conversion phase. However, as so much else has changed in catalog format/etc, then perhaps we don't need to keep this in. 

If you think it best, I'm happy to remove all other than:
```
"Star": "DeltaFunction"
```
(i.e., also throwing away the lowercase fixes and enforcing Pascal case formatting).

Reply:
Replaced:
```
replace_dict = {
    "0": "DeltaFunction",
    "star": "DeltaFunction",
    "Star": "DeltaFunction",
    "1": "Sersic",
    "galaxy": "Sersic",
    "Galaxy": "Sersic",
    "2": "Trail",
    "trail": "Trail",
    "stamp": "Stamp",
}
```

with

```
replace_dict = {"Star": "DeltaFunction"}
```